lib/query.js

```diff
@@ -1630,7 +1630,10 @@ Query.prototype.setOptions = function(options, overwrite) {
     }
   }
 
-  return Query.base.setOptions.call(this, options);
+  const optionsToSendToMQuery = Object.assign({}, options);
+  delete optionsToSendToMQuery.exec;
```

Comment:
delete is a slow operation. Checking if exec is actually set should prevent a potential performance regression
Something like

'exec' in optionsToSendGoMQuery && delete optionsToSendToMQuery.

I prefer the solution of whitelisting valid operators.

Reply:
Fixed 👍